src/Pages/Applications/DevtronApps/Details/AppConfigurations/MainContent/DeploymentTemplate/utils.tsx

```diff
@@ -0,0 +1,1221 @@
+            description: 'Something went wrong while parsing locked keys',
+        })
+    }
+
+    return currentEditorTemplateData.editorTemplate
+}
+
+export const getDeploymentTemplateInitialState = ({
+    isSuperAdmin,
+    isEnvView,
+}: GetDeploymentTemplateInitialStateParamsType): DeploymentTemplateStateType => ({
+    isLoadingInitialData: true,
+    initialLoadError: null,
+    chartDetails: {
+        charts: [],
+        chartsMetadata: {},
+        globalChartDetails: null,
+        latestAppChartRef: null,
+    },
+    publishedTemplateData: null,
+    draftTemplateData: null,
+    baseDeploymentTemplateData: null,
+    currentEditorTemplateData: null,
+    resolveScopedVariables: false,
+    isResolvingVariables: false,
+    resolvedEditorTemplate: {
+        originalTemplateString: '',
+        templateWithoutLockedKeys: '',
+    },
+    resolvedOriginalTemplate: {
+        originalTemplateString: '',
+        templateWithoutLockedKeys: '',
+    },
+    resolvedPublishedTemplate: {
+        originalTemplateString: '',
+        templateWithoutLockedKeys: '',
+    },
+    wasGuiOrHideLockedKeysEdited: false,
+    showDraftComments: false,
+    hideLockedKeys: false,
+    lockedConfigKeysWithLockType: structuredClone(DEFAULT_LOCKED_KEYS_CONFIG),
+    isSaving: false,
+    lockedDiffModalState: {
+        showLockedDiffForApproval: false,
+        showLockedTemplateDiffModal: false,
+    },
+    showSaveChangesModal: false,
+    popupNodeType: null,
+    compareFromSelectedOptionValue: CompareFromApprovalOptionsValuesType.APPROVAL_PENDING,
+    dryRunEditorMode: DryRunEditorMode.VALUES_FROM_DRAFT,
+    showDeleteOverrideDialog: false,
+    showDeleteDraftOverrideDialog: false,
+    showReadMe: false,
+    editMode: isSuperAdmin ? ConfigurationType.YAML : ConfigurationType.GUI,
+    configHeaderTab: isEnvView
+        ? CONFIG_HEADER_TAB_VALUES.OVERRIDE[0]
+        : CONFIG_HEADER_TAB_VALUES.BASE_DEPLOYMENT_TEMPLATE[0],
+    shouldMergeTemplateWithPatches: false,
+    selectedProtectionViewTab: ProtectConfigTabsType.EDIT_DRAFT,
+    isLoadingChangedChartDetails: false,
+})
+
+const handleSwitchToYAMLMode = (state: DeploymentTemplateStateType): DeploymentTemplateStateType => {
+    if (state.editMode === ConfigurationType.GUI && state.wasGuiOrHideLockedKeysEdited) {
+        try {
+            const editorTemplate = YAMLStringify(
+                applyCompareDiffOfTempFormDataOnOriginalData(
+                    YAMLStringify(state.currentEditorTemplateData.originalTemplate),
+                    state.currentEditorTemplateData.editorTemplate,
+                ),
+                { simpleKeys: true },
+            )
+
+            return {
+                ...state,
+                editMode: ConfigurationType.YAML,
+                currentEditorTemplateData: {
+                    ...state.currentEditorTemplateData,
+                    editorTemplate,
+                },
+            }
+        } catch {
+            // Do nothing
```

Comment:
can return initial state

Reply:
Can not do that since that will revert other states as well and we are returning the previous state in block below.